ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -335,32 +335,46 @@ def _router_bgp_vlans(self) -> list | None:
-                        vlan_id = int(svi["id"])
-                        vlan = {"id": vlan_id, **vlan}
-                        append_if_not_duplicate(
-                            list_of_dicts=vlans,
-                            primary_key="id",
-                            new_dict=vlan,
-                            context="BGP VLANs defined under network services",
-                            context_keys=["id", "tenant"],
-                            ignore_keys={"tenant"},
-                        )
+                sorted_vlan_list = sorted(vrf["svis"], key=self._get_vlan_aware_bundle_name_tuple_for_svis)
+                bundle_groups = itertools_groupby(sorted_vlan_list, self._get_vlan_aware_bundle_name_tuple_for_svis)
+
+                for vlan_aware_bundle_name_tuple, svis in bundle_groups:
+                    bundle_name, is_evpn_vlan_bundle = vlan_aware_bundle_name_tuple
+                    svis = list(svis)
+                    if not is_evpn_vlan_bundle:
+                        for svi in svis:
+                            if (vlan := self._router_bgp_vlans_vlan(svi, tenant, vrf)) is not None:
+                                vlan_id = int(svi["id"])
+                                vlan = {"id": vlan_id, **vlan}
+                                append_if_not_duplicate(
+                                    list_of_dicts=vlans,
+                                    primary_key="id",
+                                    new_dict=vlan,
+                                    context="BGP VLANs defined under network services",
+                                    context_keys=["id", "tenant"],
+                                    ignore_keys={"tenant"},
+                                )
 
             # L2 Vlans per Tenant
-            for l2vlan in tenant["l2vlans"]:
-                if (vlan := self._router_bgp_vlans_vlan(l2vlan, tenant, vrf={})) is not None:
-                    vlan_id = int(l2vlan["id"])
-                    vlan = {"id": vlan_id, **vlan}
-                    append_if_not_duplicate(
-                        list_of_dicts=vlans,
-                        primary_key="id",
-                        new_dict=vlan,
-                        context="BGP VLANs defined under network services",
-                        context_keys=["id", "tenant"],
-                        ignore_keys={"tenant"},
-                    )
+            sorted_vlan_list = sorted(tenant["l2vlans"], key=self._get_vlan_aware_bundle_name_tuple_for_l2vlans)
+            bundle_groups = itertools_groupby(sorted_vlan_list, self._get_vlan_aware_bundle_name_tuple_for_l2vlans)
+
+            for vlan_aware_bundle_name_tuple, l2vlans in bundle_groups:
+                bundle_name, is_evpn_vlan_bundle = vlan_aware_bundle_name_tuple
+                l2vlans = list(l2vlans)
+                if not is_evpn_vlan_bundle:
+                    for l2vlan in l2vlans:
```

Comment:
Please adjust to `continue` if `is_evpn_vlan_bundle` to make this less indented and more readable.

Reply:
same above for SVIs

Reply:
done